brion/spikeReport.cpp

```diff
@@ -198,24 +205,34 @@ std::future< Spikes > SpikeReport::readUntil( const float max )
             std::to_string( getCurrentTime( ))));
     }
 
-    if ( _impl->threadPool.hasPendingJobs() )
-    {
+    if ( _impl->busy )
         LBTHROW( std::runtime_error( "Can't read: Pending read operation" ) );
-    }
-    return _impl->threadPool.post( [&, max] { return _impl->plugin->readUntil( max ); } );
+    _impl->busy = true;
+
+    return _impl->threadPool.post(
+        [&, max]
+        {
+            auto&& spikes = _impl->plugin->readUntil( max );
+            _impl->busy = false;
+            return spikes;
+        });
 }
 
 std::future< void > SpikeReport::seek( const float toTimeStamp )
 {
     _impl->plugin->_checkNotClosed();
     if ( _impl->plugin->getAccessMode() == MODE_READ )
     {
-        if ( _impl->threadPool.hasPendingJobs() )
-        {
-            LBTHROW( std::runtime_error( "Can't seek: Pending read operation" ) );
-        }
+        if (_impl->busy)
+            LBTHROW(std::runtime_error("Can't seek: Pending read operation"));
+        _impl->busy = true;
+
         return _impl->threadPool.post(
-            [&, toTimeStamp] { return _impl->plugin->readSeek( toTimeStamp ); } );
+            [&, toTimeStamp]
+            {
+                _impl->plugin->readSeek( toTimeStamp );
```

Comment:
Seek functions, and more generally other functions, may throw.
Try to use some RAII to handle  busy properly:
struct BusyGuard 
{
BusyGuard(bool& b) : busy(b) {busy = true;}
~BusyGuard() : {busy = false;}
bool& busy;
};